include/private/gc_pmark.h

```diff
@@ -157,9 +157,10 @@ GC_INLINE mse * GC_push_obj(ptr_t obj, hdr * hhdr,  mse * mark_stack_top,
         volatile unsigned char * mark_byte_addr = \
                         (unsigned char *)(hhdr)->hb_marks + (bit_no); \
         /* Unordered atomic load and store are sufficient here. */ \
-        if (AO_char_load(mark_byte_addr) != 0) \
+        unsigned char mb = AO_char_load(mark_byte_addr); \
+        if ((mb & 1) != 0) \
```

Comment:
I assume the `1` is the bit position of the "mark bit". If so, can we make that a named constant (probably a `#define`).

Reply:
> Is it also possible to add a test (even if it's very basic) to the BWDGC test suite?

Yes I'll have a look, I've mainly been testing this through yksom so you're right it's probably worthwhile.

Reply:
In this case `1` is the bit-mask used to denote that an object is marked. I didn't make this a named constant because it is just a magic number throughout the rest of the code base and I thought it would be more confusing to only apply this to some parts. If you still think I should, I'll add it though.

Reply:
Do you mean the `1` is already used like this throughout BDWGC, before this PR?

Reply:
Got it.